MonoGame.Framework/Utilities/OS.cs

```diff
@@ -0,0 +1,46 @@
+﻿// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+namespace MonoGame.Utilities
+{
+    /// <summary>
+    /// Defines the operating system.
+    /// </summary>
+    public enum OS
```

Comment:
> That seems pretty game-specific and I'm not sure it warrants adding yet another way to check

This will come in very handy when Storage stuff gets removed(#4311).

> when there's already the platform enumeration provided by MonoGame

Again, platform and OS are two completely different things...

> and the .NET framework provides several ways to detect it at runtime.

Checking if it's running on Linux or Mac is a pain. 

Also there really is no negative effect in making the class public... the code is there regardless, it will just make it easier if someone wants to access the information.